src/ReplicaManager.ts

```diff
@@ -11,24 +11,27 @@ interface PrismaClientConstructor {
 
 type ReplicaManagerOptions = {
   clientConstructor: PrismaClientConstructor
-  replicaUrls: string[]
+  replicaUrls?: string[]
+  replicas?: PrismaClient[]
   configureCallback: ConfigureReplicaCallback | undefined
 }
 
 export class ReplicaManager {
   private _replicaClients: PrismaClient[]
 
-  constructor({ replicaUrls, clientConstructor, configureCallback }: ReplicaManagerOptions) {
-    this._replicaClients = replicaUrls.map((datasourceUrl) => {
-      const client = new clientConstructor({
-        datasourceUrl,
+  constructor({ replicaUrls = [], replicas = [], clientConstructor, configureCallback }: ReplicaManagerOptions) {
```

Comment:
Does not necessary needs to change now, but I don't think we need `configureCallback` if we allow to pass pre-configured clients now.

Reply:
Also, with what we spoke below (`replicas` and `url` being mutually exclusive, i think this type needs to reflect that too.